tfhe/src/integer/ciphertext/compressed_ciphertext_list.rs

```diff
@@ -153,46 +153,188 @@ impl CompressedCiphertextList {
 
-        let ct1 = cks.encrypt_radix(3_u32, 16);
-
-        let ct2 = cks.encrypt_signed_radix(-2, 16);
-
-        let ct3 = cks.encrypt_bool(true);
-
-        let compressed = CompressedCiphertextListBuilder::new()
-            .push(ct1)
-            .push(ct2)
-            .push(ct3)
-            .build(&compression_key);
-
-        let decompressed1 = compressed.get(0, &decompression_key).unwrap().unwrap();
-
-        let decrypted: u32 = cks.decrypt_radix(&decompressed1);
-
-        assert_eq!(decrypted, 3_u32);
-
-        let decompressed2 = compressed.get(1, &decompression_key).unwrap().unwrap();
-
-        let decrypted2: i32 = cks.decrypt_signed_radix(&decompressed2);
-
-        assert_eq!(decrypted2, -2);
-
-        let decompressed3 = compressed.get(2, &decompression_key).unwrap().unwrap();
-
-        assert!(cks.decrypt_bool(&decompressed3));
+        const MAX_NB_MESSAGES: usize = 2 * COMP_PARAM_MESSAGE_2_CARRY_2_KS_PBS_TUNIFORM_2M64
+            .packing_ks_polynomial_size
+            .0
+            / NUM_BLOCKS;
+
+        let mut rng = rand::thread_rng();
+
+        let message_modulus: u128 = cks.parameters().message_modulus().0 as u128;
+
+        for _ in 0..NB_TESTS {
+            // Unsigned
+            let modulus = message_modulus.pow(NUM_BLOCKS as u32);
+            for _ in 0..NB_OPERATOR_TESTS {
+                let nb_messages = 1 + (rng.gen::<u64>() % MAX_NB_MESSAGES as u64);
+                let messages = (0..nb_messages)
+                    .map(|_| rng.gen::<u128>() % modulus)
+                    .collect::<Vec<_>>();
+
+                let cts = messages
+                    .iter()
+                    .map(|message| cks.encrypt_radix(*message, NUM_BLOCKS))
+                    .collect_vec();
+
+                let mut builder = CompressedCiphertextListBuilder::new();
+
+                for ct in cts {
+                    let and_ct = radix_sks.bitand_parallelized(&ct, &ct);
+                    builder.push(and_ct);
+                }
+
+                let compressed = builder.build(&compression_key);
+
+                for (i, message) in messages.iter().enumerate() {
+                    let decompressed = compressed.get(i, &decompression_key).unwrap().unwrap();
+                    let decrypted: u128 = cks.decrypt_radix(&decompressed);
+                    assert_eq!(decrypted, *message);
+                }
+            }
+
+            // Signed
+            let modulus = message_modulus.pow((NUM_BLOCKS - 1) as u32) as i128;
+            for _ in 0..NB_OPERATOR_TESTS {
+                let nb_messages = 1 + (rng.gen::<u64>() % MAX_NB_MESSAGES as u64);
+                let messages = (0..nb_messages)
+                    .map(|_| rng.gen::<i128>() % modulus)
+                    .collect::<Vec<_>>();
+
+                let cts = messages
+                    .iter()
+                    .map(|message| cks.encrypt_signed_radix(*message, NUM_BLOCKS))
+                    .collect_vec();
```

Comment:
We need to compute the bitand here as well

Reply:
Of course. I forgot about the other types.